src/compiler/crystal/command.cr

```diff
@@ -334,7 +334,7 @@ class Crystal::Command
     output_format : String?,
     combine_rpath : Bool do
     def compile(output_filename = self.output_filename)
-      compiler.emit_base_filename = original_output_filename
+      compiler.emit_base_filename = output_filename.rchop(File.extname(output_filename))
```

Comment:
This is equivalent and more concise:

```suggestion
      compiler.emit_base_filename = Path[output_filename].stem
```

Reply:
This is from a [prior PR](https://github.com/crystal-lang/crystal/pull/13370). That PR actually changed the behavior of `-o` so that it also affects `--emit`; previously those emitted files would always be in the current working directory and ignore the name given to `-o`. Thus `output_filename` could now contain directory separators, and calling `Path#stem` would remove them.

Reply:
I guess the name `filename` is misleading then if it's actually a path.
But okay, don't need to fix that here.